### PR TITLE
Add ability to change payout address using cli.

### PR DESCRIPTION
**Pull request by jack60612**
Tested and working.

### CODE REVIEW

**Review by Quexington — DISMISSED** [review]
Looks like it works to me

**Review by Quexington — DISMISSED** [review]
Still looks like it works to me :+1:

**Review by mariano54 — COMMENTED** [review]
So this looks good for changing the config file. The question is, what will happen after that?

Let's say I'm farming on two machines A and B. I change the config on computer A. 
The expected behavior is that farmer A will contact the pool who will update it, which will tell B to update. Will this happen or will B complain and try to change it back?

**Comment by jack60612** [comment]
> So this looks good for changing the config file. The question is, what will happen after that?
> 
> Let's say I'm farming on two machines A and B. I change the config on computer A. The expected behavior is that farmer A will contact the pool who will update it, which will tell B to update. Will this happen or will B complain and try to change it back?

They will both try to change it back.

**Comment by mariano54** [comment]
> > So this looks good for changing the config file. The question is, what will happen after that?
> > Let's say I'm farming on two machines A and B. I change the config on computer A. The expected behavior is that farmer A will contact the pool who will update it, which will tell B to update. Will this happen or will B complain and try to change it back?
> 
> They will both try to change it back.

Maybe we should have some logic on the farmer side that if something changes on the server, we should also change it on our config file?

**Comment by mariano54** [comment]
This is not necessarily part of this PR. 
However, if we don't solve this problem, we should at least warn the user when they change it,  that they should reconfigure this in all farmer machines that they have, so they don't have mismatched. (farmer not harvester)

**Comment by jack60612** [comment]
> This is not necessarily part of this PR. However, if we don't solve this problem, we should at least warn the user when they change it, that they should reconfigure this in all farmer machines that they have, so they don't have mismatched. (farmer not harvester)

This has already been a pre existing issue, so i think i can get away with not having a warning till i fix the other issue with another pull.

**Comment by mariano54** [comment]
Good point. However I still think a just adding an extra line to the print would be good to ensure that people are aware of this issue.

**Comment by jack60612** [comment]
> Good point. However I still think a just adding an extra line to the print would be good to ensure that people are aware of this issue.

Already half way done adding it to this pull, give me a couple minutes.

**Comment by jack60612** [comment]
Done

File: chia/farmer/farmer.py [hunk 1]
```diff
@@ -496,12 +498,14 @@ async def update_pool_farmer_info() -> Tuple[Optional[GetFarmerResponse], Option
                                 farmer_info, farmer_is_known = await update_pool_farmer_info()
                                 if farmer_info is None and not farmer_is_known:
                                     self.log.error("Failed to update farmer info after POST /farmer.")
+                        if pool_config.payout_instructions.lower() != farmer_info.payout_instructions.lower():
```
Comment (mariano54) [inline chia/farmer/farmer.py:501]: famer info might be None

File: chia/farmer/farmer.py [hunk 2]
```diff
@@ -496,12 +498,14 @@ async def update_pool_farmer_info() -> Tuple[Optional[GetFarmerResponse], Option
...
+                            pool_config = replace(pool_config, payout_instructions=farmer_info.payout_instructions)
+                            just_updated = True
```
Comment (mariano54) [inline chia/farmer/farmer.py:503]: If we change the config file, won't that trigger this to be just_updated = True?

**Review by mariano54 — APPROVED** [review]

**Review by xdustinface — APPROVED** [review]

**Pull request merged**